packages/gateway/src/features/restLocation/locationHandler.ts

```diff
@@ -188,45 +188,75 @@ export async function fetchLocationHandler(
   return response
 }
 
-function batchLocationsHandler(locations: Location[]) {
-  const locationsMap = new Map(
-    locations.map((location) => [
-      location.statisticalID,
-      { ...location, uid: `urn:uuid:${uuid()}` as URNReference }
-    ])
-  )
-  const locationsBundle = {
-    resourceType: 'Bundle',
-    type: 'document',
-    entry: locations
-      .map((location) => ({
-        ...location,
-        // partOf is either Location/{statisticalID} of another location or 'Location/0'
-        partOf:
-          locationsMap.get(location.partOf.split('/')[1])?.uid ??
-          location.partOf
-      }))
-      .map(
-        (location): BundleEntry<FhirLocation> => ({
-          fullUrl: locationsMap.get(location.statisticalID)!.uid,
-          resource: {
-            ...composeFhirLocation(location),
-            ...(location.statistics && {
-              extension: generateStatisticalExtensions(location.statistics)
-            })
-          }
-        })
-      )
+function createLocationSegments(locations: Location[]): Location[][] {
+  const segments = []
+  for (const jurisdictionType of Object.keys(JurisdictionType)) {
+    const jurisdictionLocations = locations.filter(
+      (loc) => loc.jurisdictionType === jurisdictionType
+    )
+    if (jurisdictionLocations.length) {
+      segments.push(jurisdictionLocations)
+    }
+  }
+  const facilitiesOrOffices = locations.filter((loc) => !loc.jurisdictionType)
+  if (facilitiesOrOffices.length) {
+    segments.push(facilitiesOrOffices)
+  }
+  return segments
+}
+
+async function batchLocationsHandler(
+  locations: Location[],
+  h: Hapi.ResponseToolkit
+) {
+  let parentLocationsMap: Map<string, string> = new Map()
```

Comment:
Could you clarify this parentLocationMap a bit. I don't understand what it does 😞

Reply:
`parentLocationsMap` has statistical id to fhirId mapping of parent locations, I agree, I should give it more clearer name. I am thinking to change it `statisticalToFhirIDMapOfParentLocations`